ansible-tests/validations/library/validate_network_environment.py

```diff
@@ -199,6 +202,54 @@ def check_vlan_ids(vlans):
     return errors
 
 
+def check_static_ip_in_cidr(networks, static_ips):
+    '''
+    Verify that all the static IP addresses are from the corresponding network
+    range.
+    '''
+    errors = []
+    network_ranges = {}
+    # TODO(shadower): Refactor this so networks are always valid and already
+    # converted to `netaddr.IPNetwork` here. Will be useful in the other checks.
+    for name, cidr in six.iteritems(networks):
+        try:
+            network_ranges[name] = netaddr.IPNetwork(cidr)
+        except ValueError:
+            errors.append("Network '{}' has an invalid CIDR: '{}'"
+                          .format(name, cidr))
+    for role, services in six.iteritems(static_ips):
+        for service, ips in six.iteritems(services):
+            range_name = service.title().replace('_', '') + 'NetCidr'
+            if range_name in network_ranges:
+                for ip in ips:
+                    if ip not in network_ranges[range_name]:
+                        errors.append(
+                            "The IP address {} is outside of the {} range: {}"
+                            .format(ip, range_name, networks[range_name]))
+            else:
+                errors.append(
```

Comment:
This is great, but sadly the code is never exercised because the validation blows up in `check_allocation_pools_pairing` (line 129) when the service doesn't have a corresponding NetCidr :-(